loom/engine/bindings/loom/lmSQLite.cpp

```diff
@@ -0,0 +1,384 @@
+    }
+
+    int getParameterIndex(const char* name)
+    {
+        int index = sqlite3_bind_parameter_index(statementHandle, name);
+        if(name == NULL)
+        {
+            lmLogError(gSQLiteGroup, "Invalid name for getParameterIndex in database: %s", parentDB->getDBName());
+        }
+        return index;
+    }
+
+    int bindInt(int index, int value)
+    {
+        int result = sqlite3_bind_int(statementHandle, index, value);
+        if(result != SQLITE_OK)
+        {
+            lmLogError(gSQLiteGroup, "Error calling bindInt for database: %s with Result Code: %i", parentDB->getDBName(), result);
+        }
+        return result;        
+    }
+
+    int bindDouble(int index, double value)
+    {
+        int result = sqlite3_bind_double(statementHandle, index, value);
+        if(result != SQLITE_OK)
+        {
+            lmLogError(gSQLiteGroup, "Error calling bindDouble for database: %s with Result Code: %i", parentDB->getDBName(), result);
+        }
+        return result;        
+    }
+
+    int bindString(int index, const char *value)
+    {
+        int result = sqlite3_bind_text(statementHandle, index, value, -1, SQLITE_TRANSIENT); 
+        if(result != SQLITE_OK)
+        {
+            lmLogError(gSQLiteGroup, "Error calling bindString for database: %s with Result Code: %i", parentDB->getDBName(), result);
+        }
+        return result;        
+    }
+
+
+    int bindBytes(lua_State *L)
+    {
+        int index;
+        void *bytes;
+        int size;
+
+        if (!lua_isnumber(L, 1) || !lualoom_checkinstancetype(L, 2, "system.ByteArray"))
+        {
+            lmLogError(gSQLiteGroup, "Invalid parameters passed to bindBytes for database: %s", parentDB->getDBName());
+            lua_pushnumber(L, SQLITE_ERROR);
+            return 1;
+        }
+
+        //get the index we are binding to from lua
+        index = (int)lua_tonumber(L, 1);
+
+        //get our ByteArray from lua
+        utByteArray *byteArray = (utByteArray *)lualoom_getnativepointer(L, 2);
+        if(!byteArray || !byteArray->getSize())
+        {
+            bytes = NULL;
+            size = 0;
+        }
+        else
+        {
+            bytes = byteArray->getDataPtr();      
+            size = (int)byteArray->getSize();
+        }
+
+        //bind the blob to the statement
+        int result = sqlite3_bind_blob(statementHandle, index, (const void *)bytes, size, SQLITE_TRANSIENT); 
+        if(result != SQLITE_OK)
+        {
+            lmLogError(gSQLiteGroup, "Error calling bindBytes for database: %s with Result Code: %i", parentDB->getDBName(), result);
+        }
+        lua_pushnumber(L, result);
+
+        return 1;
+    }
+
+    int step()
+    {
+        int result = sqlite3_step(statementHandle); 
+        if((result == SQLITE_ERROR) || (result == SQLITE_MISUSE))
+        {
+            lmLogError(gSQLiteGroup, "Error calling bindString for database: %s with Result Code: %i", parentDB->getDBName(), result);
```

Comment:
So never do any return value checks in the cpp code? I'm fine with that if
you are. Just thought having this output here would show nice error logs
for tracking issues when people don't check their stuff :)

Easier to write and maintain the cpp without them, so your call :)
On 12 Feb 2015 5:33 AM, "Ben Garney" <notifications@github.com> wrote:

> In loom/engine/bindings/loom/lmSQLite.cpp
> <https://github.com/LoomSDK/LoomSDK/pull/261#discussion_r24556223>:
>
> > +        int result = sqlite3_bind_blob(statementHandle, index, (const void *)bytes, size, SQLITE_TRANSIENT);
> > +        if(result != SQLITE_OK)
> > +        {
> > +            lmLogError(gSQLiteGroup, "Error calling bindBytes for database: %s with Result Code: %i", parentDB->getDBName(), result);
> > +        }
> > +        lua_pushnumber(L, result);
> > +
> > +        return 1;
> > +    }
> > +
> > +    int step()
> > +    {
> > +        int result = sqlite3_step(statementHandle);
> > +        if((result == SQLITE_ERROR) || (result == SQLITE_MISUSE))
> > +        {
> > +            lmLogError(gSQLiteGroup, "Error calling bindString for database: %s with Result Code: %i", parentDB->getDBName(), result);
>
> This error seems inappropriate. I think we just want to return the error
> codes as is and let script handle it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/LoomSDK/LoomSDK/pull/261/files#r24556223>.
>